utils/telegram.js

```diff
@@ -0,0 +1,60 @@
+import fs from 'fs'
+import { Api, TelegramClient } from 'telegram'
+import { StringSession } from 'telegram/sessions/index.js'
+import random from 'random-bigint'
+import Config from '../config/index.js'
+import sendRequest from './request.js'
+
+const apiTelegram = 'https://api.telegram.org'
+const client = new TelegramClient(new StringSession(Config.get('api.session')), Number(Config.get('api.id')), Config.get('api.hash'), {})
+
+const sendMessageWithUser = async (chatId, message, replyToMsgId = null) => {
+  if (client.disconnected) await client.connect()
+  await client.invoke(
```

Comment:
CMIIW, i think the async function in best practice should be returning promises not a value/final results like this,
u can execute only when the function will be call:
so maybe like this:

```
const sendMessageWithUser = (chatId, message, replyToMsgId = null) => {
  if (client.disconnected) return client.connect()
  return client.invoke(
    new Api.messages.SendMessage({...})
  )
}
```

and u can call like this:
`await sendMessageWithUser(...)`


Async functions always return a promise. If the return value of an async function is not explicitly a promise, it will be implicitly wrapped in a promise. 
ref: 
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
- https://sutanlab.id/blog/memahami-konsep-promise-di-javascript#async---await

Reply:
yes, but for this case, you need await to connect telegram, so you need async-await

Reply:
u just need to await when the function is call, just try it!

Reply:
Oh i see, but u still can only return a promises after try connecting to telegram.
like this:
```
  if (client.disconnected) await client.connect()
  return client.invoke(
    new Api.messages.SendMessage({...})
  )
```

Reply:
oh i see, thanks insight about promises,  :partying_face: 15a2646